### PR TITLE
feat(gamma-observability): serve native kafka apis in the logs signal

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

_Jira ticket to be added (ESM observability-logs epic, story 1 — backend NATIVE support)._

## Description

Adds `NATIVE` (Kafka) API support to the Gamma observability **LOGS** signal so the ESM module's Observability > Logs screen can serve real data:

- `NATIVE` added to `SearchObservabilityLogsUseCase.LOGS_SUPPORTED_API_TYPES`, `native-kafka` added to the default entrypoint scoping — native connection documents (`v4-metrics`) are matched by the existing generic search path (they carry `entrypoint-id: native-kafka`, no separate query path needed).
- Native fields hoisted from `additional-metrics` as first-class fields on the log entry **and** the detail: `connectionStatus`, `clientId` (Kafka `client.id`), `brokerId`, `connectionDurationMs` — plus `host` (SNI) and `subscriptionId`.
- New **derived** `failureOrigin` field (`CLIENT_TO_GATEWAY` / `GATEWAY_TO_BROKER` / `GATEWAY_INTERNAL` / `UNKNOWN` / `NONE`) — the key debugging signal (which side of the proxying path failed). The gateway-written `failure-side` additional metric (see gravitee-io/gravitee-reactor-native-kafka#369) is **authoritative** when present; otherwise the Kafka error key is classified, refined by the connection status. Genuinely undecidable cases are honestly reported as `UNKNOWN` ("Undetermined") instead of being guessed. The classification vocabulary lives in one place, `NativeFailureOriginRules` (repository-api), shared by the classifier and the filter translation.
- Filters: `NATIVE_CONNECTION_STATUS` now served on LOGS, new `FAILURE_ORIGIN` filter (LOGS, NATIVE scope; per-origin boolean ES predicates preferring the explicit `failure-side`, scoped to native documents), `ERROR_KEY`/`REQUEST_ID`/`TRANSACTION_ID` extended to NATIVE. The filter validator now rejects unknown ENUM values (400) for every ENUM filter.
- Duration read falls back to the legacy `keyword_native-kafka_connection-duration-ms` key written by `gravitee-reactor-native-kafka` < 7.1 (typed-key fix in the companion reactor PR).

## Additional context

- Consumed by `gravitee-gamma-module-esm` Observability > Logs (Kafka-first columns; wire contract mirrored by the module's dev mock). Companion reactor PR: gravitee-io/gravitee-reactor-native-kafka#369 (long-typed duration + authoritative failure side) — merge order free, this PR degrades gracefully without it.
- Validated end-to-end against a local stack (master distribution + ES seeded with documents matching the exact reporter output, plus real gateway traffic): NATIVE search, all five `FAILURE_ORIGIN` values, `NATIVE_CONNECTION_STATUS`, and `/observability/filters/definition` advertising verified.
- Self-reviewed (multi-angle + verification): confirmed findings fixed in `6bafe5f05d` (unknown-enum-value 400, native-document guard on the failure-origin clause, forward-compatible known-side exclusion) and `d17be183bd`.
- **Open points for review**: (1) the analytics default entrypoint scoping was NOT widened to native — unfiltered logs totals include native connections while dashboard totals do not (noted in the `DEFAULT_ENTRYPOINT_IDS` javadoc; aligning analytics is a Gamma-team decision); (2) `StaticFilters.NATIVE_CONNECTION_STATUS` advertises a `DISCONNECTED` value the gateway never emits (pre-existing).
- Not included (follow-up): automated ES integration test against seeded native documents; `FAILURE_ORIGIN` on the ANALYTICS signal.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: gravitee-gamma/gravitee-gamma-rest-api/src/main/java/io/gravitee/gamma/rest/infra/adapter/ObservabilityLogsDataPortAdapter.java [hunk 1]
```diff
@@ -399,9 +418,67 @@ private LogEntry mapToLogEntry(BaseConnectionLog log, Map<String, ApiReference>
...
+
+    private record NativeMetrics(String connectionStatus, String clientId, String brokerId, Long connectionDurationMs) {
+        static NativeMetrics from(Map<String, Object> additionalMetrics) {
+            if (additionalMetrics == null || additionalMetrics.isEmpty()) {
+                return new NativeMetrics(null, null, null, null);
+            }
+            var duration = additionalMetrics.getOrDefault(
+                NativeApiMetricKeys.CONNECTION_DURATION_MS,
+                additionalMetrics.get(LEGACY_KEYWORD_CONNECTION_DURATION_MS)
+            );
```
[the remark this answers is not in the document]
Comment (ThibaudAV) [reply gravitee-gamma/gravitee-gamma-rest-api/src/main/java/io/gravitee/gamma/rest/infra/adapter/ObservabilityLogsDataPortAdapter.java]: Addressed in 5dfdefb178.

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java [hunk 2]
```diff
@@ -90,6 +95,10 @@ private static JsonObject buildElasticQuery(MetricsQuery.Filter filter) {
 
         addMcpProxyPromptsFilter(filter, mustFilterList);
 
+        addNativeConnectionStatusesFilter(filter, mustFilterList);
+
+        addFailureOriginsFilter(filter, mustFilterList);
+
         if (!mustFilterList.isEmpty()) {
             return JsonObject.of("bool", JsonObject.of("must", JsonArray.of(mustFilterList.toArray())));
         }
```
[the remark this answers is not in the document]
Comment (ThibaudAV) [reply gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java:105]: Addressed in 5dfdefb178.

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java [hunk 3]
```diff
@@ -242,6 +251,160 @@ private static void addMcpProxyPromptsFilter(MetricsQuery.Filter filter, List<Js
...
+            var predicate = failureOriginPredicate(origin);
+            if (predicate != null) {
+                perOrigin.add(predicate);
+            }
+        }
+        if (!perOrigin.isEmpty()) {
+            mustFilterList.add(
+                JsonObject.of("bool", JsonObject.of("should", JsonArray.of(perOrigin.toArray()), "minimum_should_match", 1))
+            );
+        }
```
[the remark this answers is not in the document]
Comment (ThibaudAV) [reply gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java:297]: Addressed in 5dfdefb178.

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java [hunk 4]
```diff
@@ -242,6 +251,160 @@ private static void addMcpProxyPromptsFilter(MetricsQuery.Filter filter, List<Js
...
+        branches.add(
+            JsonObject.of(
+                "terms",
+                JsonObject.of(RequestV2MetricsV4Fields.ERROR_KEY, NativeFailureOriginRules.BROKER_SIDE_ERROR_KEYS.toArray())
+            )
+        );
+        for (String prefix : NativeFailureOriginRules.BROKER_SIDE_ERROR_KEY_PREFIXES) {
+            branches.add(JsonObject.of("prefix", JsonObject.of(RequestV2MetricsV4Fields.ERROR_KEY, prefix)));
+        }
+        return JsonObject.of("bool", JsonObject.of("should", JsonArray.of(branches.toArray()), "minimum_should_match", 1));
```
[the remark this answers is not in the document]
Comment (ThibaudAV) [reply gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchMetricsQueryAdapter.java]: Addressed in 5dfdefb178.

**Review by a-cordier — APPROVED** [review]

**Review by callaertanthony — APPROVED** [review]
Reviewed the full diff and traced the load-bearing parts. High-quality, heavily-tested change — approving. A few non-blocking notes below.

Verified: `HOST`/`SUBSCRIPTION_ID` resolve via static import from `RequestV2MetricsV4Fields`; `SearchLogsFilters` is `@Builder(toBuilder=true)` with no positional constructors, so the two new record components don't break callers; the display classifier and the ES filter predicate agree across the corner cases (side-wins-over-key, `UNKNOWN_SERVER_ERROR` + `CONNECTION_ERROR`, keyless + internal, `DISCONNECTED`); field mapping is pinned by the ES fixture test.

### Non-blocking

**1. The classification is implemented twice, not shared (structural).**
`FailureOriginClassifier.classify` (display) and `SearchMetricsQueryAdapter.failureOriginPredicate` (ES filter) independently re-encode the same decision tree. Only the vocabulary (`NativeFailureOriginRules`) is shared — the ordering and phase-fallback logic is duplicated. They agree today and each is well-tested in isolation, but no single test asserts equivalence, so a future edit to one can silently drift from the other. Cheapest insurance: one table-driven test that runs representative `(errorKey, connectionStatus, failureSide)` tuples through the classifier and asserts each row is matched by exactly the filter predicate for its origin.

**2. Orphaned Javadoc — `ObservabilityLogsDataPortAdapter` (~L1141).**
The Javadoc describing the `NativeMetrics` record floats above the `LEGACY_KEYWORD_CONNECTION_DURATION_MS` constant, which has its own Javadoc immediately after — so the constant is double-documented and the record is undocumented. Move the `NativeMetrics` doc down onto the record declaration.

**3. `failure-side` present + `connection-status` absent diverges (edge case).**
The display path maps a recognized `failureSide` to an origin regardless of connection status; the ES `FAILURE_ORIGIN` filter's native-document guard requires `exists(connection-status)`. Such a row would show a `failureOrigin` but be invisible to the filter. Almost certainly unreachable since the gateway writes both — a one-line confirmation is enough, not a fix.

### For awareness (not defects)
- `validateEnumValues` now rejects unknown ENUM values with 400 for every ENUM filter, not just the new ones — a deliberate, cross-cutting behavior change (previously such values were silently dropped downstream).
- The two open points already called out in the description stand: analytics default entrypoint scope not widened to native (mixed-env logs total ≠ dashboard total), and `NATIVE_CONNECTION_STATUS` advertising a `DISCONNECTED` value the gateway never emits (pre-existing).
